src/components/builder.ts

```diff
@@ -48,24 +48,68 @@ export class Builder {
     }
 
     /**
-     * Kill the current building process.
+     * Stop the current build process.
      */
-    kill() {
-        const proc = this.currentProcess
-        if (proc) {
-            const pid = proc.pid
+    async stop() {
+        if (this.currentProcess) {
+            const envVarsPosix: ProcessEnv = {PATH: '/bin:/usr/bin'}
+            const envVarsWindows: ProcessEnv = {PATH: '%SystemRoot%\\system32;%SystemRoot%;%SystemRoot%\\System32\\Wbem'}
+
+            const currentProcessExit = new Promise(listener => this.currentProcess?.on('exit', listener))
+            const pid = this.currentProcess.pid
+
+            // Try to terminate the build process gracefully first.
             if (process.platform === 'linux' || process.platform === 'darwin') {
-                cp.execSync(`pkill -P ${pid}`)
+                try {
+                    // Terminate build process and all its child processes.
+                    await this.spawnAndWait('pkill', ['-TERM', '-P', pid.toString()], {env: envVarsPosix})
+                    await this.spawnAndWait('kill', ['-TERM', pid.toString()], {env: envVarsPosix})
```

Comment:
This is exactly what `this.currentProcess.kill()` does. Why not use it?

Reply:
The second line, you mean? It is more consistent with the other `pkill` and `taskkill` commands (in that it runs a program), but I do see your point. I can switch to using `kill` if you stll prefer.